doc/tutorial/argument_parser/index.md

```diff
@@ -456,3 +456,87 @@ Add a seqan3::regex_validator to the first positional option that expects the `f
 \solution
 \snippet doc/tutorial/argument_parser/solution6.cpp file_validator
 \endsolution
+
+
+# The Version Update Feature
+
+SeqAn applications that are using the seqan3::argument_parser can check SeqAn servers for version updates.
+The functionality helps getting new versions out to users faster.
+It is also used to inform application developers of new versions of the SeqAn library
+which means that applications ship with less bugs.
+For privacy implications, please see: https://github.com/seqan/seqan3/wiki/Update-Notifications
+
+### Customized Messages
+
+The version information you receive depends on whether you are an application user or developer.
+We differentiate this by inquiring the `NDEBUG` (no-debug) macro.
+
+Case 1: `NDEBUG` is set.
+It's the default in our application and represents that you are a user.
+The only message you will eventually encounter is the following:
+
+```
+[APP INFO] :: There is a newer version of this application available.
+[APP INFO] :: If this app is developed by SeqAn, visit www.seqan.de for updates.
+[APP INFO] :: If you don't want to receive this message again set --version_check 0
+```
+
+Case 2: `NDEBUG` is NOT set.
+If you build one of our application or your own one in debug mode, we will consider you as a developer.
+Therefore we will inform you whenever a new library version is available:
+
+```
+[SEQAN INFO] :: There is a newer SeqAn version available!
+[SEQAN INFO] :: Please visit www.seqan.de for an update or inform the developer of this app.
+[SEQAN INFO] :: If you don't want to recieve this message again set --version-check 0
+```
+If you are working on your own application, using the SeqAn ArgumentParser,
+we will inform you about the possibility to register your application with us.
+This will make the distribution of your application version simple and convenient.
```

Comment:
"If the application is not registered with the server, you will be prompted to register it."